### PR TITLE
Mishal blog fetch

### PR DESCRIPTION
**Pull request by misusonu18**
# Added Blog From The Hashnode Directly Using API

## Video 

https://user-images.githubusercontent.com/52093984/158189006-567c02bb-941d-4f59-9159-18b072c538f3.mp4

### CODE REVIEW

File: js/custom_script.js [hunk 1]
```diff
@@ -254,4 +254,31 @@ $(function () {
...
+                            <div class="blog-cover-image-div">
+                                <img src="`+ element.coverImage + `">
+                            </div>
+
+                            <div class="blog-title-div">
+                                <h5>`+ element.title +`</h5>
+                                <span>`+ element.brief + `</span>
+                            </div>
+                        </a>
+                    </li>
```
Comment (hirenkeraliya) [inline js/custom_script.js:278]: use template tag for this kind of thing.

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

**Comment by hirenkeraliya** [comment]
Header link is directly to the blog website instead of this make scroll to the blog content.

File: index.html [hunk 2]
```diff
@@ -2040,6 +2040,48 @@ <h4>Maybe you?</h4>
...
+                            <button class="form-btn">
+                                <img src="images/know-more-icon.png" alt="Read all">
+                            </button>
+                        </label>
+                    </a>
+                </div>
+
+                <div class="team-right">
+                    <ul class="team-list" id="blog-ul">
+                        <template id="blog-display">
```
Comment (hirenkeraliya) [inline index.html:2068]:
```diff
-blog-display
+blog-item
```

File: netlify.toml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[functions]
+    directory = "functions"
+
```
Comment (hirenkeraliya) [inline netlify.toml:3]: why blank space?

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: js/custom_script.js [hunk 4]
```diff
@@ -256,6 +256,24 @@ $(function () {
     });
 });
 
+axios.post('/.netlify/functions/fetch_blogs').then((result) => {
+    let blogUl = document.getElementById('blog-ul');
+    const articles = result.data.data.user.publication.posts;
```
Comment (hirenkeraliya) [inline js/custom_script.js:261]: is there any way to fetch only 2 blog posts?

File: js/custom_script.js [hunk 5]
```diff
@@ -256,6 +256,24 @@ $(function () {
...
+    const blogDisplayTemplate = document.getElementById('blog-item');
+
+    articles.forEach((element, index) => {
+        const blogDisplayCard = blogDisplayTemplate.content.cloneNode(true);
+        if (index < 2) {
+            blogDisplayCard.querySelector('.blog-link').href = "https://blog.freshbits.in/" + element.slug;
+            blogDisplayCard.querySelector('.blog-cover-image').src = element.coverImage;
+            blogDisplayCard.querySelector('.blog-cover-image').alt = element.slug;
```
Comment (hirenkeraliya) [inline js/custom_script.js:269]: element.title

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: js/custom_script.js (hunk 4 above)
Comment (misusonu18) [reply js/custom_script.js:261]: not possible, by default gets 6 data

**Review by hirenkeraliya — APPROVED** [review]

File: js/custom_script.js [hunk 6]
```diff
@@ -256,6 +256,24 @@ $(function () {
     });
 });
 
+axios.post('/.netlify/functions/fetch_blogs').then((result) => {
+    let blogUl = document.getElementById('blog-ul');
```
Comment (gauravmak) [inline js/custom_script.js:260]: const

File: js/custom_script.js [hunk 7]
```diff
@@ -256,6 +256,24 @@ $(function () {
...
+    const articles = result.data.data.user.publication.posts;
+    const blogDisplayTemplate = document.getElementById('blog-item');
+
+    articles.forEach((element, index) => {
+        const blogDisplayCard = blogDisplayTemplate.content.cloneNode(true);
+        if (index < 2) {
+            blogDisplayCard.querySelector('.blog-link').href = "https://blog.freshbits.in/" + element.slug;
```
Comment (gauravmak) [inline js/custom_script.js:267]: Are they providing full url?

File: js/custom_script.js [hunk 8]
```diff
@@ -256,6 +256,24 @@ $(function () {
     });
 });
 
+axios.post('/.netlify/functions/fetch_blogs').then((result) => {
```
Comment (gauravmak) [inline js/custom_script.js:259]: Document this part in the readme file.

```md
# Blog articles

- We are fetching blog articles from the Hadhnode API which is open for all.
- The ajax request is made on the page load to the Netlify function in the `functions/fetch_blogs.js` file.
- Last two blog articles are displayed on the home page.
```

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: js/custom_script.js (hunk 7 above)
Comment (misusonu18) [reply js/custom_script.js:267]: no

**Pull request merged**